drivers/digital-io/max22190/max22190.c

```diff
@@ -0,0 +1,492 @@
+	if (ret)
+		return ret;
+
+	*enabled = no_os_field_get(MAX22190_FAULT2_WBE_MASK, reg_val);
+
+	return 0;
+}
+
+/**
+ * @brief Set Wire Break detection for requested channel
+ * @param desc - MAX22190 device descriptor.
+ * @param ch - Channel to set the WBE to.
+ * @param enabled - State of the WBE for the specific channel to be set.
+ * @return 0 in case of succes, negative error code otherwise.
+*/
+int max22190_wbe_set(struct max22190_desc *desc, uint32_t ch, bool enabled)
+{
+	if (ch > MAX22190_CHANNELS - 1)
+		return -EINVAL;
+
+	return max22190_reg_update(desc, MAX22190_FILTER_IN_REG(ch),
+				   MAX22190_FAULT2_WBE_MASK,
+				   no_os_field_prep(MAX22190_FAULT2_WBE_MASK,
+						   enabled));
+}
+
+/**
+ * @brief Get fault enable from the fault registers
+ * @param desc - MAX22190 device descriptor
+ * @param fault_enable - Specific fault enable to read
+ * @param enabled - State of the selected enable (0 for disabled, 1 for enabled)
+ * @return 0 in case of succes, negative error code otherwise
+*/
+int max22190_fault_enable_get(struct max22190_desc *desc,
+			      enum max22190_fault_enable fault_enable,
+			      bool *enabled)
+{
+	int ret;
+	uint32_t reg_val;
+
+	switch(fault_enable) {
+	case MAX22190_FAULT1_WBGE ... MAX22190_FAULT1_FAULT2E:
+		ret = max22190_reg_read(desc, MAX22190_FAULT1_EN_REG, &reg_val);
+		if (ret)
+			return ret;
+		*enabled = no_os_field_get(MAX22190_FAULT_MASK(fault_enable),
+					   reg_val);
+		break;
+	case MAX22190_FAULT2_RFWBSE ... MAX22190_FAULT2_FAULT8CKE:
+		ret = max22190_reg_read(desc, MAX22190_FAULT2_EN_REG, &reg_val);
+		if (ret)
+			return ret;
+		*enabled = no_os_field_get(MAX22190_FAULT_MASK(fault_enable -
+					   MAX22190_FAULT2_RFWBSE), reg_val);
+		break;
+	case MAX22190_CFG_24VF:
+		ret = max22190_reg_read(desc, MAX22190_CONFIG_REG, &reg_val);
+		if (ret)
+			return ret;
+		*enabled = no_os_field_get(MAX22190_CFG_24VF_MASK, reg_val);
+		break;
+	case MAX22190_CFG_REFDI_SH_ENA:
+		ret = max22190_reg_read(desc, MAX22190_CONFIG_REG, &reg_val);
+		if (ret)
+			return ret;
+		*enabled = no_os_field_get(MAX22190_CFG_REFDI_MASK, reg_val);
+		break;
+	default:
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+/**
+ * @brief Set fault enable in the fault registers
+ * @param desc - MAX22190 device descriptor
+ * @param fault_enable - Specific fault enable to write
+ * @param enabled - State of the selected enable (0 for disabled, 1 for enabled)
+ * @return 0 in case of succes, negative error code otherwise
+*/
+int max22190_fault_enable_set(struct max22190_desc *desc,
+			      enum max22190_fault_enable fault_enable,
+			      bool enabled)
+{
+	int ret;
+
+	switch(fault_enable) {
+	case MAX22190_FAULT1_WBGE ... MAX22190_FAULT1_FAULT2E:
+		return max22190_reg_update(desc, MAX22190_FAULT1_EN_REG,
+					   MAX22190_FAULT_MASK(fault_enable),
+					   no_os_field_prep(MAX22190_FAULT_MASK(fault_enable),
+							   enabled));
+	case MAX22190_FAULT2_RFWBSE ... MAX22190_FAULT2_FAULT8CKE:
```

Comment:
so what if I want to enabled MAX22190_FAULT2_RFWBSE ?
`MAX22190_FAULT1_FAULT2E` should probably be automatically enabled by the driver, it is currently not

Reply:
This would imply keeping track of fault2 enables somewhere in desc->fault2en or something like that.
When any fault2 enable gets enabled, `MAX22190_FAULT1_FAULT2E` should be set.
When no fault2 enable remains, `MAX22190_FAULT1_FAULT2E` should be cleared.

Reply:
I'll do that but having a fault2en in the desc isn't necessary.